onnxruntime/contrib_ops/cpu/transformers/greedy_search_impl_base.h

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <vector>
+#include "contrib_ops/cpu/transformers/generation_shared.h"
+#include "contrib_ops/cpu/transformers/generate_impl_base.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+namespace transformers {
+
+template <typename T>
+struct GreedySearchState : public IGreedySearchState<T> {
+  Sequences sequences;
+
+  void Init(AllocatorPtr cpu_allocator,
+            AllocatorPtr allocator,
+            int batch_size,
+            int vocab_size,
+            int sequence_length,
+            int max_length,
+            bool /*is_cuda*/) {
+    // below buffers are on cpu
+    this->sequences_space = AllocateBuffer<int32_t>(cpu_allocator,
+                                                    sequences_space_buffer_,
+                                                    SafeInt<size_t>(2) * batch_size * max_length);
+    memset(this->sequences_space.data(), 0, this->sequences_space.size_bytes());
+    this->sequences.Init(this->sequences_space, static_cast<int>(batch_size), sequence_length, max_length);
+
+    this->sequence_lengths = AllocateBuffer<int32_t>(cpu_allocator, sequence_lengths_buffer_, batch_size);
+    this->eos_meet = AllocateBuffer<bool>(cpu_allocator, eos_meet_buffer_, batch_size);
+    memset(this->eos_meet.data(), 0, this->eos_meet.size_bytes());
+
+    this->next_tokens_cpu = AllocateBuffer<int64_t>(cpu_allocator,
+                                                    next_tokens_cpu_buffer_,
+                                                    SafeInt<size_t>(batch_size));
+    this->next_tokens = AllocateBuffer<int32_t>(cpu_allocator, next_tokens_buffer_, SafeInt<size_t>(batch_size));
```

Comment:
what's relation between next_tokens and next_tokens_cpu as both are in CPU?

Reply:
next_tokens is the int32 version of next_tokens_cpu